src/comments/comments.service.ts

```diff
@@ -6,28 +6,51 @@ import { Comment } from './entities/comment.entity';
 import { Post } from "../posts/entities/post.entity";
 import mongoose from "mongoose";
 import { Profile } from "../profiles/entities/profile.entity";
+import { Notification } from "../notifications/enitities/notification.entity";
+import { User } from "../user/entity/User";
+import { UserService } from "../user/user.service";
 
 @Injectable()
 export class CommentsService {
   constructor(@Inject('COMMENT_MODEL') private readonly commentModel: Model<Comment>,
               @Inject('POST_MODEL') private readonly postModel: Model<Post>,
-              @Inject('PROFILE_MODEL') private readonly profileModel: Model<Profile>) {}
+              @Inject('PROFILE_MODEL') private readonly profileModel: Model<Profile>,
+              @Inject('NOTIFICATION_MODEL') private readonly  notificationModel: Model<Notification>,
 
+             ) {}
+
+  /*
+  here we will also add it to the notifications list
+   */
   async create(createCommentDto: CreateCommentDto) {
     const post = await this.postModel
       .findOne({ _id: createCommentDto.postId })
       .exec();
+
+    var  a =  post.username;
+    const postOwner = await this.profileModel.findOne({a}).exec();
+
     const profile = await this.profileModel
       .findOne({ userId: createCommentDto.userId })
       .exec();
 
-    if (post != null && profile != null) {
+    if (post != null && profile != null ) {
       post.comments.push({
         profile: profile.id,
         username: profile.username,
         date: createCommentDto.date,
         text: createCommentDto.text,
       });
+
+      this.notificationModel.create({
```

Comment:
I think you should wrap it in an if which is checking if postOwner is not null, for better error handling :)